client/web/src/components/Timeline.tsx

```diff
@@ -0,0 +1,62 @@
+import classNames from 'classnames'
+import { formatDistance } from 'date-fns/esm'
+import React, { FunctionComponent, ReactNode } from 'react'
+import { Timestamp } from './time/Timestamp'
+
+export interface TimelineStage {
+    icon: ReactNode
+    text: ReactNode
+    date?: string | null
+    className?: string
+}
+
+export interface TimelineProps {
+    stages: TimelineStage[]
+    now?: () => Date
+    className?: string
+}
+
+export const Timeline: FunctionComponent<TimelineProps> = ({ stages, now, className }) => (
+    <>
+        <h3>Timeline</h3>
+
+        <div className={className}>
+            {stages.map((stage, stageIndex) => {
+                const previousDate = stages
+                    .map(stage => stage.date)
+                    .filter((date, index) => !!date && index < stageIndex)
+                    .pop()
+
+                return (
+                    stage.date && (
+                        <>
+                            {previousDate && (
+                                <div className="d-flex align-items-center">
+                                    <div className="flex-0">
+                                        <div className="executor-timeline-task-separator" />
```

Comment:
Following [BEM](http://getbem.com/naming/)
```suggestion
                                        <div className="timeline__executor-timeline-task-separator" />
```

Reply:
Can you re-check to see if my changes here are correct? If not I can follow up in a subsequent PR.

Reply:
Looks good!